tests/test_utils_template.py

```diff
@@ -1 +1,62 @@
+import scrapy
+from os.path import exists, join, abspath, dirname
+from shutil import rmtree, copytree, move
+from tempfile import mkdtemp
+from twisted.trial import unittest
+from scrapy.utils.template import get_template_dir
+
 __doctests__ = ['scrapy.utils.template']
+
+
+class getTemplateDirTest(unittest.TestCase):
+
+    def setUp(self):
+        self.settings = {}
+        self.settings['TEMPLATES_DIR_BASE'] = [
+                        abspath(join(dirname(__file__), '..', 'templates')),
+                        join(scrapy.__path__[0], "templates")
+                    ]
+        self.settings['TEMPLATES_PROJECT_BASE'] = 'project'
+        self.settings['TEMPLATES_SPIDERS_BASE'] = 'spiders'
+        self.proj_path = mkdtemp()
+        copytree(self.settings['TEMPLATES_DIR_BASE'][1], join(self.proj_path, "templates"))
+        self.template_folder = join(self.proj_path, "templates")
+
+    def tearDown(self):
+        rmtree(self.proj_path)
+
+    def test_standard(self, template_type="project"):
+        option1 = join(self.settings['TEMPLATES_DIR_BASE'][0], template_type)
+        option2 = join(self.settings['TEMPLATES_DIR_BASE'][1], template_type)
+        if(exists(option1)):
```

Comment:
I have tried to implement this. Couldn't come up with a better method than as shown below :/ . Kindly let me know if your views.